### PR TITLE
clean up dataset conversion readme

### PR DESCRIPTION
**Pull request by codestar12**
Cleans up existing readme and adds finetuning dataset example

### CODE REVIEW

File: scripts/data_prep/README.md [hunk 1]
```diff
@@ -12,21 +12,48 @@ Using the `convert_dataset_hf.py` script...
...
-  --concat_tokens 2048 --tokenizer EleutherAI/gpt-neox-20b --eos_text '<|endoftext|>'
+  --concat_tokens 2048 --tokenizer EleutherAI/gpt-neox-20b --eos_text '<|endoftext|>' \
+  --compression zstd
 ```
 
 Using the `convert_dataset_json.py` script...
 
 <!--pytest.mark.skip-->
 ```bash
 # Convert C4 dataset to StreamingDataset format
```
Comment (dakinggg) [inline scripts/data_prep/README.md:23]: comment should be updated?

File: scripts/data_prep/README.md [hunk 2]
```diff
@@ -12,21 +12,48 @@ Using the `convert_dataset_hf.py` script...
...
-  --concat_tokens 2048 --tokenizer EleutherAI/gpt-neox-20b --eos_text '<|endoftext|>'
+  --concat_tokens 2048 --tokenizer EleutherAI/gpt-neox-20b --eos_text '<|endoftext|>' \
+  --compression zstd
 ```
 
 Where `--path` can be a single json file, or a folder containing json files, and split the intended split (hf defaults to train).
 
 ## Converting a finetuning dataset
-Using the `convert_finetuning_dataset.py` script...
+Using the `convert_finetuning_dataset.py` script you can run a commandsuch as:
```
Comment (dakinggg) [inline scripts/data_prep/README.md:34]:
```suggestion
Using the `convert_finetuning_dataset.py` script you can run a command such as:
```

File: scripts/data_prep/README.md [hunk 3]
```diff
@@ -12,21 +12,48 @@ Using the `convert_dataset_hf.py` script...
...
+
+```bash
+python convert_finetuning_dataset.py --dataset ""Muennighoff/P3"" \
```
Comment (dakinggg) [inline scripts/data_prep/README.md:37]: whats up with the double double quotes?

File: scripts/data_prep/README.md [hunk 4]
```diff
@@ -12,21 +12,48 @@ Using the `convert_dataset_hf.py` script...
...
+
+- You are running the script in the terminal or command prompt where `python` command is recognized.
+- `"Muennighoff/P3"` is the dataset you want to convert. Substitute "Muennighoff/P3" with the name or path of your dataset.
+- `train` and `validation` are the splits of the dataset to convert.
+- `llmfoundry.data.finetuning.tasks:p3_preprocessing_function` is a string that provides the name or import path of the function used to preprocess the dataset. Substitute it with your actual preprocessor.
+- `s3://<bucket>/muennighoff-p3` is the root path of your output directory where MDS shards will be stored. Replace this with the actual path to your output directory.
+
+Please note that you need to fill in actual values for "your_preprocessing_function" and "/path/to/your/output_directory" in the command above for it to work correctly.
+
+Also, if you want to keep a local copy of the output when out_root is remote, you can use the `--local` argument:
```
Comment (dakinggg) [inline scripts/data_prep/README.md:53]:
```suggestion
Also, if you want to keep a local copy of the output when `out_root` is remote, you can use the `--local` argument:
```

File: scripts/data_prep/README.md [hunk 5]
```diff
@@ -12,21 +12,48 @@ Using the `convert_dataset_hf.py` script...
...
+--preprocessor "your_preprocessing_function"\
+ --out_root "/path/to/your/output_directory"
+```
+
+This example assumes:
+
+- You are running the script in the terminal or command prompt where `python` command is recognized.
+- `"Muennighoff/P3"` is the dataset you want to convert. Substitute "Muennighoff/P3" with the name or path of your dataset.
+- `train` and `validation` are the splits of the dataset to convert.
+- `llmfoundry.data.finetuning.tasks:p3_preprocessing_function` is a string that provides the name or import path of the function used to preprocess the dataset. Substitute it with your actual preprocessor.
```
Comment (dakinggg) [inline scripts/data_prep/README.md:48]: i think there are more details on how to define a preprocessing function somewhere else, maybe point to those?

File: scripts/data_prep/README.md [hunk 6]
```diff
@@ -12,21 +12,48 @@ Using the `convert_dataset_hf.py` script...
...
+  --compression zstd
 ```
 
 Where `--path` can be a single json file, or a folder containing json files, and split the intended split (hf defaults to train).
 
 ## Converting a finetuning dataset
-Using the `convert_finetuning_dataset.py` script...
+Using the `convert_finetuning_dataset.py` script you can run a commandsuch as:
+
+```bash
```
Comment (dakinggg) [inline scripts/data_prep/README.md:36]: you'll probably want to pytest skip the bash commands to get CI to pass

File: scripts/data_prep/README.md (hunk 1 above)
Comment (codestar12) [reply scripts/data_prep/README.md:23]: yes ... not compressing is a big mistake ... lots of wasted space and engress

File: scripts/data_prep/README.md (hunk 3 above)
Comment (codestar12) [reply scripts/data_prep/README.md:37]: oh thats just a mistake

File: scripts/data_prep/README.md (hunk 5 above)
Comment (alextrott16) [reply scripts/data_prep/README.md:48]: Reviewing this had led me to catch some issues with preprocessing. I'll address that in a separate PR, along with other improvements.

File: scripts/data_prep/README.md (hunk 1 above)
Comment (dakinggg) [reply scripts/data_prep/README.md:23]: sorry, I meant the `# Convert C4 dataset to StreamingDataset format` comment doesn't seem to be correct for this command

**Review by alextrott16 — COMMENTED** [review]
The finetuning stuff looks good to me, so I'm approving that part. But, as mentioned in a comment. I have caught some QOL stuff with the preprocessing function that I think needs to be addressed in a separate PR. That will require me to edit the finetuning section, but what you added here is a very useful starting point for that.

File: scripts/data_prep/README.md (hunk 5 above)
Comment (codestar12) [reply scripts/data_prep/README.md]: @dakinggg  I updated with a link to relevant files, but unless I missed it I can't find pointers on how to define the function (other than the code)

**Review by dakinggg — APPROVED** [review]
Could you add a link in the top level description to streaming?

**Pull request merged**